.github/workflows/publish.yml

```
@@ -96,14 +107,29 @@ jobs:
         run: |
           docker buildx build -f $env:DOCKERFILE -o build .
       - name: upload build
-        if: github.event_name == 'repository_dispatch'
+        if: github.event_name == 'workflow_dispatch'
         shell: bash
         env:
-          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+          RELEASE_TAG: ${{ needs.create_release.outputs.release_id }}
           PLATFORM: ${{ matrix.platform }}
           GH_TOKEN: ${{ github.token }}
         working-directory: build
         run: |
           chmod +x "bmx"
           tar -czvf "bmx-$PLATFORM.tar.gz" "bmx"
           gh release upload "$RELEASE_TAG" "bmx-$PLATFORM.tar.gz"
+
+  publish_release:
+    needs: [create_release, build, build_docker]
+    if: |
+      ${{ github.event_name == 'workflow_dispatch' && github.event.inputs.dry_run == false }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    env:
+      RELEASE_TAG: ${{ needs.create_release.outputs.release_id }}
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - name: finalize release
+        run: |
+          # gh release edit "$env:RELEASE_ID" --draft=false
```

not powershell (I know this is commented out for now)
```suggestion
          # gh release edit "$RELEASE_ID" --draft=false
```